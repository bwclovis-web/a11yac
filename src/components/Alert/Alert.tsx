import React, {useContext, useEffect, useState} from 'react'
import styled from 'styled-components'

import PageContext from '../PageProvider/PageProvder';

interface AlertInterface {
    alertMessage: string
}

interface AlertStylesI {
    active?: boolean,
    theme?: string
}

const AlertStyles = styled.div<AlertStylesI>`
    position: absolute;
    top:  0;
    padding: 20px;
    border-bottom: 4px solid ${props => props.theme === 'dark' ? 'var(--grey-lt)' : 'var(--blue-lt)'};
    border-left: 1px solid var(--grey);
    border-right: 1px solid var(--grey);
    box-shadow: 0 0 10px 2px rgba(0,0,0, 0.3);
    width: 90%;
    z-index: 2;
    text-align: center;
    transform: ${props => props.active ? `translateY(0)` : `translateY(-10vw)` };
    opacity: ${props => props.active ? 1 : 0};
    transition: all .3s ease-in-out;
`

const AlertMessage: React.FC<AlertInterface> = ({alertMessage}) => {
    const {clicked, theme} = useContext(PageContext)
    const [showAlert, setShowAlert] = useState(false);
    const [message, setMessage] = useState('');

    useEffect(() => {
        clicked && handleDisplayAlert();
    }, [clicked]);

    const handleDisplayAlert = () => {
        setShowAlert(true);
        setMessage(alertMessage);
        setTimeout(() => {
            setShowAlert(false);
            setMessage('')
        }, 2000)
    }

    return (
        <AlertStyles active={showAlert} theme={theme}>
            <p className="h3" role="alert">
                {message}
            </p>
        </AlertStyles>
    )
}

export default AlertMessage;