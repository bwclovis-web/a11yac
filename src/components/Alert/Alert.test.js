import React from 'react';
import renderer from 'react-test-renderer';
import Alert from './Alert';

describe("Alert", () => {
    it('should render correctly ', () => {
        const tree = renderer.create(<Alert/>).toJSON()
        expect(tree).toMatchSnapshot()
    });
})