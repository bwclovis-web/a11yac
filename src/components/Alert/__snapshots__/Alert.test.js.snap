// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly  1`] = `
<div
  className="sc-bdfBwQ gHRuNF"
>
  <p
    className="h3"
    role="alert"
  >
    
  </p>
</div>
`;
